### PR TITLE
Adding NSM NSE Connection Request code

### PR DESCRIPTION
**Pull request by sbezverk**
Adding initial code for NSM to NSE Connection Request communication.

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

### CODE REVIEW

File: cmd/nse/nse.go [hunk 1]
```diff
@@ -161,3 +260,18 @@ func socketCleanup(listenEndpoint string) error {
 	}
 	return nil
 }
+
+func getCurrentNS() string {
+	buf := make([]byte, MaxSymLink)
+	numBytes, err := syscall.Readlink(netnsfile, buf)
```
Comment (fkautz) [inline cmd/nse/nse.go]: Why syscall.Readlink over os.Readlink?

File: cmd/nse/nse.go [hunk 2]
```diff
@@ -161,3 +260,18 @@ func socketCleanup(listenEndpoint string) error {
...
+	if err != nil {
+		return ""
+	}
+	link := string(buf[0:numBytes])
+	nsRegExp := regexp.MustCompile("net:\\[(.*)\\]")
+	submatches := nsRegExp.FindStringSubmatch(link)
+	if len(submatches) >= 1 {
+		return submatches[1]
+	}
+	return ""
```
Comment (fkautz) [inline cmd/nse/nse.go]: Should we return an empty string here, or return an error?

File: plugins/nsmserver/nsmserver_client.go [hunk 3]
```diff
@@ -322,13 +418,15 @@ func startClientServer(id string, endpoints *nsmClientEndpoints) {
 	}
 
 	unix.Umask(socketMask)
-	sock, err := net.Listen("unix", listenEndpoint)
+
+	sock, err := newCustomListener(listenEndpoint) // net.Listen("unix", listenEndpoint)
```
Comment (fkautz) [inline plugins/nsmserver/nsmserver_client.go]: Any reason we are keeping this comment around?

File: plugins/nsmserver/nsmserver_client.go [hunk 4]
```diff
@@ -322,13 +418,15 @@ func startClientServer(id string, endpoints *nsmClientEndpoints) {
...
 	if err != nil {
 		logger.Errorf("failure to listen on socket %s with error: %+v", client.socketPath, err)
 		client.allocated = false
 		return
 	}
-	grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+
+	grpcServer := grpc.NewServer( /*grpc.UnaryInterceptor(grpcUnaryIntercept)*/ )
```
Comment (fkautz) [inline plugins/nsmserver/nsmserver_client.go]: What is this comment?

File: cmd/nse/nse.go (hunk 1 above)
Comment (edwarnicke) [reply cmd/nse/nse.go]: os.Readlink returns an error and *not* the content of the link if the link is broken.

/proc/self/ns/net is a link, but its *always* a broken link.  The netns inode is in the link content... os.Readlink doesn't work.

Comment (sbezverk) [reply cmd/nse/nse.go]: right

File: plugins/nsmserver/nsmserver_client.go (hunk 3 above)
Comment (sbezverk) [reply plugins/nsmserver/nsmserver_client.go]: will remove, thanks for catching it.

File: plugins/nsmserver/nsmserver_client.go (hunk 4 above)
Comment (sbezverk) [reply plugins/nsmserver/nsmserver_client.go]: I was fighting with grpc, so it is a leftover of that fight. will remove.

File: cmd/nse/nse.go (hunk 2 above)
Comment (sbezverk) [reply cmd/nse/nse.go]: I have no strong opinion here, if you feel this func signature should be `(string, error)`, I can change it.

**Pull request merged**